src/tests/Microsoft.PowerFx.Interpreter.Tests/DisplayNameTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerFx.Core;
+using Microsoft.PowerFx.Core.Public.Types;
+using Microsoft.PowerFx.Core.Public.Values;
+using Microsoft.PowerFx.Core.Utils;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Microsoft.PowerFx.Interpreter.Tests
+{    
+    public class DisplayNameTests 
```

Comment:
Is there a test for DisabledDisplayNameProvider?

Reply:
Would also be good to add a test for the immutability issues we discussed. 

r = new RecordType() 

r1 = r.Add()  // "Logical" --> "Display1"
r2 = r.Add()  // "Logical" --> "Display2"

Ensure that r2 doesn't interfere with r1.